sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -51,7 +51,8 @@ using Azure::Identity::_detail::TokenCache;
 using Azure::Identity::_detail::TokenCredentialImpl;
 
 namespace {
-std::string const MsgPrefix = "Identity: AzureCliCredential";
+std::string const CredentialName = "AzureCliCredential";
+std::string const MsgPrefix = "Identity: " + CredentialName;
```

Comment:
The pattern around MsgPrefix is inconsistent in all credentials.  Why have it at all though?  We could shrink the size of the credential if we didn't use it.

Does starting with `Identity:` help?  In which case there should be a single Log function that we funnel identity through which adds the Prefix.

Reply:
I like the idea of single place where the prefix gets appended.

I think we should have prefixes - for instance, with `DefaultAzureCredential`, when something fails, users want to figure out, and also suppose they are not very experienced users, so they go to the logs as we suggest. But logs have every HTTP request, and some other stuff. Especially now, until we reviewed all the logging code, some things from the other places in the SDK probably shouldn't even be logged; log message format is different from place to place. But anyways, back to the point - we can tell the user "To diagnose `DefaultAzureCredential`, enable logs, and look at the messages that start with `Identity:`" without the prefix, it will be hard to diagnose.

So, I'll update this PR to add a single location for prefix.

Reply:
Let me to actually do that in a subsequent PR, so that the changes are easier to review.

Reply:
#4551